Update crashhandler.py to solve #11042

slel opened the pull request:
Tell users to provide information about their operating system
and how they installed the application.

slel committed:
Update crashhandler.py to solve #11042

Tell users to provide information about their operating system
and how they installed the application.

jdemeyer commented:
Info about operating system is already in the crash log. What's mostly missing is *context*: what did users do to end up with that crash log?

slel committed:
Improve instructions on reporting crash

Tell users who want to email a crash report to add
- information about operating system
- information on how they installed
- information on how to reproduce the crash

takluyver commented on IPython/core/crashhandler.py:
```diff
@@ -53,6 +53,16 @@
+- information about your operating system: is it Linux, macOS, Windows, other,
+  and which version, for example Ubuntu 16.04, macOS 10.13.2, Windows 10 Pro,
+  is it 32-bit or 64-bit, ...
+- information on how you installed {app_name}: from binary, compiled from source,
```
'compiled from source' doesn't really apply to pure Python projects like IPython. I'd give examples like "using pip or conda, from GitHub, or as part of a Docker container."

takluyver commented on IPython/core/crashhandler.py:
```diff
@@ -53,6 +53,16 @@
+  is it 32-bit or 64-bit, ...
+- information on how you installed {app_name}: from binary, compiled from source,
+  via Docker, on a virtual machine...
+- information on how to reproduce the crash: what exact sequence of instructions
```
Since all three points start with "information on", let's phrase the bit at the top "Please also include information on:" and cut those words from the bullet points.

Also, I'd like the bullet points to start with capital letters  ("How to reproduce...").

takluyver commented on IPython/core/crashhandler.py:
```diff
@@ -53,6 +53,16 @@
+Please also include in your email:
+- information about your operating system: is it Linux, macOS, Windows, other,
+  and which version, for example Ubuntu 16.04, macOS 10.13.2, Windows 10 Pro,
+  is it 32-bit or 64-bit, ...
```
Ending a paragraph with `...` to signify that these are examples seems a bit too casual for the tone of this message. Let's find a way to make it read a bit better.

slel committed:
Make crash handler tell users to email more detail

Pull request #11043, to solve issue #11042.
Taking into account reviewer comments on previous draft.

takluyver commented:
Thanks!
